File: frontend/src/app/_models/student.ts
import { Subject } from "./subject"

export class Student {
    public id:  string = ''
    public name: string = ''
    public neptun: string = ''
    public sumcredit: number = 0
    public image: string = ''
    public Subjects: Array<Subject> = []

    public createSubjects(subjectList: Array<any>) {
        if (subjectList) {
          subjectList.map((x: any) => {
            let subject = new Subject();
            subject.id = x.id;
            subject.name = x.name;
            subject.neptun = x.neptun;
            subject.credit = x.credit;
            subject.exam = x.exam;
            subject.image = x.image;
            this.Subjects.push(subject);
          });
        }
      }
}

File: backend/backend/Models/Student.cs
using Microsoft.AspNetCore.Identity;
using Newtonsoft.Json;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace backend.Models
{
    public class Student
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public string Id { get; set; }
        public string? Name { get; set; }
        public string? Neptun { get; set; }
        private int? sumCredit;

        public int? SumCredit
        {
            get
            {
                int? sum = 0;
                if (Subjects is not null)
                {
                    foreach (var item in Subjects)
                    {
                        sum += item.Credit;
                    }
                }
                return sum;
            }
        }
        public string? Image { get; set; }

        [NotMapped]
        public virtual ICollection<Subject>? Subjects { get; set; }

        public Student()
        {
            Subjects = new HashSet<Subject>();
        }
    }
}
